peak/mapper/formula_constructor.py

```diff
@@ -0,0 +1,67 @@
+import operator
+from functools import partial, reduce, lru_cache
+
+or_reduce = partial(reduce, operator.or_)
+and_reduce = partial(reduce, operator.and_)
+
+
+class FormulaConstructor:
+    indent = "|   "
+
+    @staticmethod
+    def _to_smt(v):
```

Comment:
If you aren't going to override static methods in sub classes there is no reason for them to be static methods and not free functions.

Reply:
More accurately named `_to_hwtypes`